Add a note about UI test entry limitation (#1019)
rust-lang/rustc-dev-guide: src/tests/adding.md

[hw-main]: https://github.com/rust-lang/rust/blob/master/src/test/ui/hello_world/main.rs
[hw]: https://github.com/rust-lang/rust/blob/master/src/test/ui/hello_world/

We now have a ton of UI tests and some directories have too many entries.
This is a problem because it isn't editor/IDE friendly and GitHub UI won't
show more than 1000 entries. To resolve it and organize semantic structure,
we have a tidy check to ensure the number of entries is less than 1000.
However, since `src/test/ui` (UI test root directory) and
`src/test/ui/issues` directories have more than 1000 entries,
we set a different limit for each directories. So, please

> avoid putting a new test there as possible and try to find a more relevant place.
> For example, if your test is related to closure, you should put it to

Reviewer: ```suggestion
avoid putting a new test there and try to find a more relevant place.
For example, if your test is related to closures, you should put it in
```

Author: Thanks! Addressed it.
